### PR TITLE
TG-1016 Update README

### PR DESCRIPTION
**Pull request by pauloxnet**

### CODE REVIEW

**Review by trottomv — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -6,27 +6,31 @@
...
 
 ### 🐋 Docker
 
 In order to run Talos via Docker, a working [Docker installation](https://docs.docker.com/get-docker/) is the only requirement.
 
-### 👨‍💻 Shell command
+### 🐍 Virtual environment
 
-In order to run Talos as a shell command, first clone the repository in a local projects directory
+In order to run Talos in a virtual environment first clone the repository in a local projects directory and the enter in it:
```
Comment (niccolomineo) [inline README.md:17]:
```suggestion
In order to run Talos in a virtual environment, first clone the repository in a local projects directory and ensure it is your current root:
```

File: README.md [hunk 2]
```diff
@@ -6,27 +6,31 @@
...
+python3.12 -m venv .venv
+. .venv/bin/activate
+python -m pip install -r requirements/common.txt
+```
 
-| Requirements           | Instructions                                                                 |
-| ---------------------- | ---------------------------------------------------------------------------- |
-| 🌎 Terraform           | [Install Guide](https://learn.hashicorp.com/tutorials/terraform/install-cli) |
-| 🐍 Python Dependencies | `pip install -r talos-django/requirements/common.txt`                        |
+Optionally, in order to use `terraform`, you have to install it using the [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```
Comment (niccolomineo) [inline README.md:33]:
```suggestion
You can leverage the optional `terraform` integration by installing the package via the official [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```

**Review by trottomv — DISMISSED** [review]

File: bootstrap/runner.py [hunk 3]
```diff
@@ -275,8 +275,8 @@ def format_files(self):
             [
                 "python3",
                 "-m",
-                "black",
-                "-q",
+                "ruff",
+                "format",
```
Comment (niccolomineo) [inline bootstrap/runner.py:279]: We said we'd leave out the boostraps from upgrading to `ruff format` because of that bug in `ruff` itself.

File: README.md [hunk 4]
```diff
@@ -6,29 +6,34 @@
...
 
-Then, install the following requirements
+Then, create and activate a virtual environment and install the requirements:
 
-| Requirements           | Instructions                                                                 |
-| ---------------------- | ---------------------------------------------------------------------------- |
-| 🌎 Terraform           | [Install Guide](https://learn.hashicorp.com/tutorials/terraform/install-cli) |
-| 🐍 Python Dependencies | `pip install -r talos-django/requirements/common.txt`                        |
+```console
+python3.12 -m venv .venv
```
Comment (niccolomineo) [inline README.md:28]:
```suggestion
python3 -m venv .venv
```

File: README.md [hunk 5]
```diff
@@ -49,12 +54,13 @@ cd ~/projects
 ### 🐋 Docker
 
 ```console
-docker run --interactive --tty --rm --volume $PWD/.dumps:/app/.dumps --volume $PWD/.logs:/app/.logs --volume $PWD:/data 20tab/talos-django:latest
+docker run --interactive --tty --rm --volume $PWD/.talos_dumps:/app/.dumps --volume $PWD/.talos_logs:/app/.logs --volume $PWD:/data 20tab/talos-django:latest
```
Comment (niccolomineo) [inline README.md:57]: I would reuse the same dirs locally so as to keep the dumps and logs in "buckets" the user would expect to find such data.

File: README.md [hunk 6]
```diff
@@ -6,29 +6,34 @@
...
-| 🌎 Terraform           | [Install Guide](https://learn.hashicorp.com/tutorials/terraform/install-cli) |
-| 🐍 Python Dependencies | `pip install -r talos-django/requirements/common.txt`                        |
+```console
+python3.12 -m venv .venv
+source .venv/bin/activate
+python3 -m pip install --upgrade pip setuptools
+python3 -m pip install -r requirements/common.txt
+```
+
+You can leverage the optional `terraform` integration by installing the `cli` via the official [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```
Comment (niccolomineo) [inline README.md:34]:
```suggestion
You can leverage the optional `terraform` integration by installing the cli package via the official [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```

Comment (trottomv) [inline README.md:34]: I would consider the installation of Terraform to be 'mandatory' and not optional. The only scenario where it is not required is in generating a project solely locally, with no intention of setting up remote services and environments.
I would rephrase the instructions by removing the optional aspect.

File: README.md (hunk 5 above)
Comment (trottomv) [reply README.md:57]: In the `~/projects` directory where it is suggested to execute the setup with `docker run` there is no context related to the execution of talos. When mounting the two volumes, two unintelligible hidden directories `~/projects/.logs` and `~/projects/.dumps` are generated. 
I consider a result like `~/projects/.talos_logs` and `~/projects/.talos_dumps` to be clearer and more understandable.

File: bootstrap/runner.py (hunk 3 above)
Comment (trottomv) [reply bootstrap/runner.py:279]: It seems to be a misconfiguration of our requirements rather than a bug in Ruff itself.

File: README.md (hunk 4 above)
Comment (pauloxnet) [reply README.md:28]: Using `python3` can be an alias for old and incompatible Python version.

File: README.md (hunk 5 above)
Comment (niccolomineo) [reply README.md:57]: Moreover, that would be more garbage added to my system. I really do not like that. :D 

Moreover:

Why unintelligible? Mappings to `.dumps` and `.logs` are in the command itself, the user cannot get mistaken in any way.

`talos_dumps` and `talos_logs` are hidden too, so that's not a valid argument either.

File: README.md (hunk 6 above)
Comment (pauloxnet) [reply README.md:34]: As you said, the repo can be used without `terraform` in certain condition.
Please add a suggestion that specify the conditions that require using 'terrfaorm'.

File: bootstrap/runner.py (hunk 3 above)
Comment (pauloxnet) [reply bootstrap/runner.py:279]: `ruff` here is used to format the newly generated project code and not for the repo code.

Comment (niccolomineo) [reply bootstrap/runner.py:279]: > It seems to be a misconfiguration of our requirements rather than a bug in `ruff` itself. It was necessary to move `ruff` into the `commons.in`

It's a confirmed bug by Astral.

File: README.md (hunk 5 above)
Comment (pauloxnet) [inline README.md:57]:
```suggestion
docker run --interactive --tty --rm --volume $PWD/.dumps:/app/.dumps --volume $PWD/.logs:/app/.logs --volume $PWD:/data 20tab/talos-django:latest
```

File: README.md [hunk 7]
```diff
@@ -6,29 +6,34 @@
...
-| 🌎 Terraform           | [Install Guide](https://learn.hashicorp.com/tutorials/terraform/install-cli) |
-| 🐍 Python Dependencies | `pip install -r talos-django/requirements/common.txt`                        |
+```console
+python3.12 -m venv .venv
+source .venv/bin/activate
+python3 -m pip install --upgrade pip setuptools
+python3 -m pip install -r requirements/common.txt
+```
+
+You can leverage the optional `terraform` integration by installing the cli package via the official [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```
Comment (pauloxnet) [inline README.md:34]:
```suggestion
The `terraform` cli package is required, unless you want to generate a project only locally. To install it we suggest to use the official [install guide](https://learn.hashicorp.com/tutorials/terraform/install-cli).
```

**Review by niccolomineo — APPROVED** [review]
